internal/services/admin/server.go

```diff
@@ -203,6 +206,207 @@ func (a *AdminServiceImpl) TriggerWorkflow(ctx context.Context, req *contracts.T
 	}, nil
 }
 
+func getOpts(ctx context.Context, req *contracts.TriggerWorkflowRequest, a *AdminServiceImpl) (*repository.CreateWorkflowRunOpts, error) {
+	tenant := ctx.Value("tenant").(*dbsqlc.Tenant)
+	tenantId := sqlchelpers.UUIDToStr(tenant.ID)
+
+	createContext, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	isParentTriggered := req.ParentId != nil
+
+	if isParentTriggered {
+		if req.ParentStepRunId == nil {
+			return nil, status.Error(
+				codes.InvalidArgument,
+				"parent step run id is required when parent id is provided",
+			)
+		}
+
+		if req.ChildIndex == nil {
+			return nil, status.Error(
+				codes.InvalidArgument,
+				"child index is required when parent id is provided",
+			)
+		}
+
+		// Unsure about this piece
```

Comment:
@abelanger5 @grutt 

Here we seem to be checking to see if we have already created a child workflow run and returning that if we have? 

How should this work for bulk workflow run triggers? 

Should I just return new workflow runs
Should I record this workflow_run_id and return it with the others?

Reply:
Yes, I'd record and return it with the others, that seems sensible to me. We'll need to find a way to associate it properly on the client-side, but we'll deal with that when we get there.

Reply:
sounds good